### PR TITLE
[Tools] Updating: vswhere.exe and premake5.exe

### PR DESCRIPTION
**Pull request by Xphalnos**
Updated dependencies for compilation.

| **exe** | **Old** | **New** |
| --------- | --------- | ---------- |
| **vswhere.exe** | **2.6.7** | **3.1.7** |
| **Premake5.exe** | **Premake5-beta1** | **Premake5-beta2** |

### CODE REVIEW

**Comment by Gliniak** [comment]
I would leave updating directly stored executables to repository owner due to some security concerns

File: .github/workflows/Windows_build.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
         uses: actions/setup-python@v5
         if: matrix.vsver == '2019'
         with:
-          python-version: '3.12'
+          python-version: '3.13'
```
Comment (Margen67) [inline .github/workflows/Windows_build.yml]: 3.13 isn't on `windows-2022` yet: https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#python

**Comment by Margen67** [comment]
> I would leave updating directly stored executables to repository owner due to some security concerns

I checked the file hashes and they match the official releases.

**Comment by Xphalnos** [comment]
It's ok @Margen67 I've upgraded to Python 3.12

**Pull request merged**